### DISCUSSION TITLE
Compare to grblHAL BoB by Barrett, T41U5XBB?

### DISCUSSION THREAD

**Discussion by stevempotter** [Expatria-Technologies/grblhal_2000_PrintNC#5] [General]
Phil Barrett has made a similar CNC controller that uses Teensy 4.1, [available on Tindie](https://www.tindie.com/products/philba/grblhal-bob-unkit-for-teensy-41-t41u5xbb/?pt=ac_prod_search). I am using it with my MPCNC and it works well. Can you please mention how the grblhal_2000 compares to that one? I see your's has easy Raspberry Pi integration -- is that for Ocotoprint or what did you have in mind?
Also, Do you plan to make it available soon on Tindie or similar?

**Comment by andrewmarles** [comment]
Sorry for the late replies, I was not getting notifications on these discussion items (no @andrewmarles)

Just looking quickly at top-line items, Hal2000 has the following differences:
1) Integrated 5V and 12V supplies
2) Support for closed-loop drivers
3) dual limit inputs on each axis - with 12V power integrated
4) RS485 support integrated
5) QWIIC interface if you can find PCA9615 (hopefully back in stock in 2022).
6) Plug and play support for RGB plugin board (just connect 8 pin connector)
7) RPI connector.

Basically just made the board I wanted for my machines.

**Reply by stevempotter** [reply]
Thanks for the info! BTW, you can tick the All Activity option in the Watch pull-down at upper right to be sure you get notified of things posted to your repository. 
I will be eager to see how a QWIIC pendant is implemented.

**Reply by andrewmarles** [reply]
Yeah, I have the QWIIC pendant working on my machines.  I use a Pi Pico and it works with the I2C keypad plugin.  Hopefully will have something worthy of publishing in the near future.

![image](https://user-images.githubusercontent.com/6061539/146828262-e4f5b63e-8008-4453-a0b3-33954fd80062.png)

**Reply by stevempotter** [reply]
That QWIIC pendant looks lovely! Please do share the details when they are available.

**Comment by andrewmarles** [comment]
Jogger has gone public

https://github.com/Expatria-Technologies/I2C_Jog_Controller
